shipyard_blueprints/rudderstack/shipyard_rudderstack/cli/trigger_sync_cli.py

```diff
@@ -20,17 +22,32 @@ def main():
 
     rudderstack = RudderStackClient(access_token=access_token)
     # execute trigger sync
-    rudderstack.trigger_sync(source_id)
+    try:
+        rudderstack.trigger_sync(source_id)
+    except ExitCodeException as e:
+        rudderstack.logger.error(e.message)
+        sys.exit(e.exit_code)
+
     # create artifacts folder to save run id
-    base_folder_name = shipyard.logs.determine_base_artifact_folder(
-        'rudderstack')
-    artifact_subfolder_paths = shipyard.logs.determine_artifact_subfolders(
-        base_folder_name)
-    shipyard.logs.create_artifacts_folders(artifact_subfolder_paths)
-
-    # save sync run id as variable
-    shipyard.logs.create_pickle_file(artifact_subfolder_paths,
-                                     'source_id', source_id)
+    if args.wait_for_completion == "TRUE":
+        poke = args.poke_interval
+        if 0 < int(poke) <= 60:
+            rudderstack.logger.info(f"Setting poke interval to {poke} minutes")
+        else:
+            rudderstack.logger.error("Poke interval must be between 1 and 60 minutes")
+            sys.exit(rudderstack.EXIT_CODE_INVALID_POKE_INTERVAL)
```

Comment:
Let's handle exit codes in the same way that you did for some of the others, by using that custom class in the templates package

Reply:
had it outside of the class since it is more of an application limitation